### ISSUE TITLE
Move CommonServiceLocator DI to seperate project

### ISSUE THREAD

**Issue by gjunge** [SolrNet/SolrNet#302]
In order to remove any dependencies from the SolrNet project, I suggest moving all files dealing with the Default DI (Startup.cs, IContainer.cs and Container.cs) to another project.

- When another DI  is used, we do not need the default DI included.
- CommonServicesLocator is not a netstandard2.0 class, it gives warnings.

In order not to make any breaking changes for the end users I suggest the following:

- The NuGet ID for SolrNet will be changed to `SolrNet.Core`. 
- All DI integration projects now reference this new `SolrNet.Core package`
- The new CommonServicesLocator.SolrNet project, will create a NuGet file which has the ID `SolrNet`.

So basically for the users using the `SolrNet` NuGet package nothing will change (except they will now have an extra dll for CommonServicesLocator.SolrNet)
For other DI users, they will not have an unnecessary dependency on CommonServiceLocator.

**Comment by xavier-morera** [comment]
Agreed @gjunge . Also I believe many people use other DI besides the default one.

**Comment by xavier-morera** [comment]
Completed, thanks to @gjunge
